functions/go/export-terraform/terraformgenerator/resources.go

```diff
@@ -148,8 +175,23 @@ func (resource *terraformResource) GetOrganization() *terraformResource {
 }
 
 type referencePath struct {
-	kind string // If kind is unset, it will be auto-detected
-	path []string
+	kind            string // If kind is unset, it will be auto-detected
+	path            []string
+	customRetriever func(*sdk.KubeObject) string // custom func to retrieve ref from KubeObject
+}
+
+// singleCACustomRetriever creates a customRetriever for a compute address ref
+func singleCACustomRetriever(path []string) func(*sdk.KubeObject) string {
```

Comment:
My first impression of CA in the function name was "certificate authority" which this is not related to. Can this be called `getCustomComputeAddress` or something similar?

Reply:
Updated to use `singleComputeAddressRetriever` instead